src/Arcus.Testing.Tests.Integration/Integration/DataFactory/RunDataFlowTests.cs

```diff
@@ -114,6 +115,86 @@ public async Task RunDataFlow_WithCsvFileOnSourceAndDataSetParameter_SucceedsByG
+            };
+            var dataFlowParametersWithTypes = new Dictionary<string, string>()
+            {
+                { "StringDataFlowParam", "string" },
+                { "BoolDataFlowParam", "boolean" },
+                { "IntDataFlowParam", "integer" }
+            };
+            var dataFlowParametersWithValues = new Dictionary<string, dynamic>
+            {
+                { "StringDataFlowParam", $"'{Bogus.Lorem.Word()}'" },
+                { "BoolDataFlowParam", $"{Bogus.Random.Bool().ToString().ToLowerInvariant()}()" },
+                { "IntDataFlowParam", Bogus.Random.Int(0, int.MaxValue) }
+            };
+            await using var dataFlow = await TemporaryDataFactoryDataFlow.CreateWithCsvSinkSourceAsync(
+                Configuration,
+                Logger,
+                ConfigureCsv,
+                tempDataFlowOptions: dataFlowOptions =>
+                {
+                    dataFlowOptions.Source.AddFolderPathParameters(sourceDataSetParameterKeyValues);
+                },
+                dataFlowParameters: dataFlowParametersWithTypes);
+
+            string expectedCsv = GenerateCsv();
+            await dataFlow.UploadToSourceAsync(expectedCsv, sourceDataSetParameterKeyValues.Select(d => d.Value).ToArray());
+
+            // Act
+            DataFlowRunResult result = await _session.Value.RunDataFlowAsync(dataFlow.Name, dataFlow.SinkName, options =>
+            {
+                foreach (var sourceDataSetParameter in sourceDataSetParameterKeyValues)
+                {
+                    options.AddDataSetParameter(dataFlow.SourceName, sourceDataSetParameter.Key, sourceDataSetParameter.Value);
+                }
+                foreach (var param in dataFlowParametersWithValues)
+                {
+                    options.AddDataFlowParameter(param.Key, param.Value);
+                }
+            });
+
+            // Assert
+            CsvTable expected = AssertCsv.Load(expectedCsv, ConfigureCsv);
+            var actualData = result.GetDataAsCsv(ConfigureCsv);
+            AssertCsv.Equal(expected, actualData, options =>
+            {
+                foreach (var dataFlowParam in dataFlowParametersWithTypes)
+                {
+                    options.IgnoreColumn(dataFlowParam.Key);
+                }
+            });
+
+            // Create new CSV with just expected columns and values from dataflow params
+            var lineKeys = string.Join(';', dataFlowParametersWithTypes.Keys);
+            var lineValues = string.Join(';', dataFlowParametersWithValues.Values).Replace("\'", string.Empty).Replace("(", string.Empty).Replace(")", string.Empty);
+            var expectedString = $"{lineKeys}{Environment.NewLine}";
+            // Generate as many lines as we have in actualData
+            for (var i = 0; i < actualData.RowCount; i++)
+            {
+                expectedString += $"{lineValues}{Environment.NewLine}";
+            }
```

Comment:
```suggestion
            var lineKeys = string.Join(';', dataFlowParametersWithTypes.Keys);
            var lineValues = string.Join(';', dataFlowParametersWithValues.Values).Replace("\'", string.Empty).Replace("(", string.Empty).Replace(")", string.Empty);
            var expectedString = $"{lineKeys}{Environment.NewLine}";
            // Generate as many lines as we have in actualData
            for (var i = 0; i < actualData.RowCount; i++)
            {
                expectedString += $"{lineValues}{Environment.NewLine}";
            }
```
**requires change:** instead of a code comment to explain the code, maybe we should extract this to a `private static` method instead, as it obscures the test body's intent. It could even return a `CsvTable`.